### PR TITLE
[BugFix] Fix create table from empty select bug

### PR DESCRIPTION
**Pull request by xiaoyong-z**
Signed-off-by: xyz <a997647204@gmail.com>

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #https://github.com/StarRocks/StarRocksTest/issues/1060

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

In the current implementation, creating a table from empty select will fail. This is because in #10232, we use _num_rows_load_total both for source_node and sink_node. For example, for broker load/insert into select query, _num_rows_load_total is used to count the load rows of the source nodes. In some cases, load rows from source_node and sink_node are different, which introduces some bugs.

In this pr, we use two variables _num_rows_load_source_total and _num_rows_load_sink_total to distinguish between the number of loaded rows for source_node and sink_node respectively.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 42 / 76 (55.26%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadJob.java|0|1|00.00%|[224]|
|:large_blue_circle:|com/starrocks/load/loadv2/BrokerLoadJob.java|0|2|00.00%|[365, 367]|
|:large_blue_circle:|com/starrocks/load/loadv2/InsertLoadJob.java|2|6|33.33%|[136, 138, 139, 142]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadManager.java|1|2|50.00%|[609]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|2|50.00%|[2211]|
|:large_blue_circle:|com/starrocks/load/EtlStatus.java|35|60|58.33%|[360, 361, 362, 373, 375, 377, 378, 379, 381, 385, 386, 387, 389, 393, 394, 395, 397, 422, 423, 424, 425, 468, 496, 499, 502]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 42 / 76 (55.26%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadJob.java|0|1|00.00%|[224]|
|:large_blue_circle:|com/starrocks/load/loadv2/BrokerLoadJob.java|0|2|00.00%|[365, 367]|
|:large_blue_circle:|com/starrocks/load/loadv2/InsertLoadJob.java|2|6|33.33%|[136, 138, 139, 142]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadManager.java|1|2|50.00%|[608]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|2|50.00%|[2211]|
|:large_blue_circle:|com/starrocks/load/EtlStatus.java|35|60|58.33%|[360, 361, 362, 373, 375, 377, 378, 379, 381, 385, 386, 387, 389, 393, 394, 395, 397, 422, 423, 424, 425, 468, 496, 499, 502]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 42 / 76 (55.26%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadJob.java|0|1|00.00%|[224]|
|:large_blue_circle:|com/starrocks/load/loadv2/BrokerLoadJob.java|0|2|00.00%|[365, 367]|
|:large_blue_circle:|com/starrocks/load/loadv2/InsertLoadJob.java|2|6|33.33%|[136, 138, 139, 142]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadManager.java|1|2|50.00%|[608]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|2|50.00%|[2211]|
|:large_blue_circle:|com/starrocks/load/EtlStatus.java|35|60|58.33%|[360, 361, 362, 373, 375, 377, 378, 379, 381, 385, 386, 387, 389, 393, 394, 395, 397, 422, 423, 424, 425, 468, 496, 499, 502]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|3|3|100.00%|[]|

**Comment by xiaoyong-z** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 42 / 76 (55.26%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadJob.java|0|1|00.00%|[224]|
|:large_blue_circle:|com/starrocks/load/loadv2/BrokerLoadJob.java|0|2|00.00%|[365, 367]|
|:large_blue_circle:|com/starrocks/load/loadv2/InsertLoadJob.java|2|6|33.33%|[136, 138, 139, 142]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadManager.java|1|2|50.00%|[608]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|2|50.00%|[2211]|
|:large_blue_circle:|com/starrocks/load/EtlStatus.java|35|60|58.33%|[360, 361, 362, 373, 375, 377, 378, 379, 381, 385, 386, 387, 389, 393, 394, 395, 397, 422, 423, 424, 425, 468, 496, 499, 502]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: gensrc/thrift/FrontendService.thrift [hunk 1]
```diff
@@ -440,7 +440,11 @@ struct TReportExecStatusParams {
 
   16: optional i64 backend_id
 
-  17: optional i64 loaded_bytes
+  17: optional i64 sink_load_bytes
```
Comment (chaoyli) [inline gensrc/thrift/FrontendService.thrift:443]: Does sink_load_rows is necessary?

File: be/src/runtime/runtime_state.h [hunk 2]
```diff
@@ -359,12 +379,16 @@ class RuntimeState {
 
     // put here to collect files??
     std::vector<std::string> _output_files;
-    std::atomic<int64_t> _num_rows_load_total{0};      // total rows read from source
+    std::atomic<int64_t> _num_rows_load_source_total{0};  // total rows read from load source
+    std::atomic<int64_t> _num_bytes_load_source_total{0}; // total bytes read from load source
+
+    std::atomic<int64_t> _num_rows_load_sink_total{0};  // total rows read from load sink
```
Comment (meegoo) [inline be/src/runtime/runtime_state.h:385]: I think the name `_num_rows_load_sink_total` is little confuse. and can you add some comment explain the difference between `_num_rows_load_source_total` and `_num_rows_load_sink_total`, why need it.

File: be/src/runtime/runtime_state.h [hunk 3]
```diff
@@ -359,12 +379,16 @@ class RuntimeState {
 
     // put here to collect files??
     std::vector<std::string> _output_files;
-    std::atomic<int64_t> _num_rows_load_total{0};      // total rows read from source
+    std::atomic<int64_t> _num_rows_load_source_total{0};  // total rows read from load source
```
Comment (meegoo) [inline be/src/runtime/runtime_state.h:382]: _num_rows_load_from_source?

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by xiaoyong-z** [comment]
run starrocks_fe_unittest

**Comment by xiaoyong-z** [comment]
run starrocks_fe_unittest

**Review by meegoo — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by xiaoyong-z** [comment]
run starrocks_fe_unittest

**Comment by xiaoyong-z** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 42 / 76 (55.26%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadJob.java|0|1|00.00%|[224]|
|:large_blue_circle:|com/starrocks/load/loadv2/BrokerLoadJob.java|0|2|00.00%|[365, 367]|
|:large_blue_circle:|com/starrocks/load/loadv2/InsertLoadJob.java|2|6|33.33%|[136, 138, 139, 142]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadManager.java|1|2|50.00%|[608]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|2|50.00%|[2211]|
|:large_blue_circle:|com/starrocks/load/EtlStatus.java|35|60|58.33%|[360, 361, 362, 373, 375, 377, 378, 379, 381, 385, 386, 387, 389, 393, 394, 395, 397, 422, 423, 424, 425, 468, 496, 499, 502]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|3|3|100.00%|[]|

**Pull request merged**